### DISCUSSION TITLE
Issue with Log Printing Format in Allure Report

### DISCUSSION THREAD

**Discussion by santhoshshetty** [allure-framework/allure2#2049] [Questions & Support] [answered]
Hello everyone,

I'm currently facing an issue with the log printing format in the Allure report after migrating to SLF4J with Log4j2 implementation. I would appreciate any assistance or suggestions to resolve this problem. In SLF4J2 we just use the placeholders - {} and the dymanic values are resolved defaultly.

In our project, we use the following method to print log statements in the Allure report:

@Step("INFO: {0}")
public void logInfo(final String format, final Object... argArray) {
    log.info(format, argArray);
}

For example, if we call logInfo("Value is {}", "sample"), we expect the log message in the Allure report to be displayed as "Value is sample". However, the actual log message shown in the Allure report is "Value is {}".

When we expand the statement, we can see the format and argArray details correctly. Our goal is to resolve this issue and achieve the expected result of the log printing in the Allure report in the desired format ("Value is sample").

Input - logInfo("Value is {}", "sample")
Expected Result - Value is sample
Actual Result - Value is {}

I would greatly appreciate any insights, suggestions, or possible solutions to address this problem. Thank you in advance for your assistance.

![image](https://github.com/allure-framework/allure2/assets/4597235/2c67c44f-f668-431e-bb06-1095b8f08d5f)

Best regards,
Santhosh

**Comment by baev** [comment, marked as answer]
I would suggest you use `Allure.step()` API instead of `@Step` annotation. That way, you'll get full control over generated step names.
